### PR TITLE
Directory structure and readme edits

### LINKED ISSUE #51: Improved Canopy-App Directory Structure
Need for improved directory structure for Canopy-App source codes (src), example inputs (data), docs, etc.

### PR DESCRIPTION
**Pull request by zmoon**
* [x] `src` -- Makefile and Fortran sources, flat structure within for now
* [x] `input` -- sample input datasets *and* sample namelist
* [x] `output` -- where we want the output files to be written to
* [x] `docs` -- separate Markdown file(s) featuring some of the details from the readme, in preparation for HTML docs someday
  - ended up adding some sections to the readme and doing a bit of formatting/organizing instead
* [x] logo in readme
* ~~initial CMake config~~ (for a next PR)

Resolves #51

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,30 +1,42 @@
...
  - `DEBUG  = 0(off; default) or DEBUG  =1(on)`
  - `NETCDF = 0(off) or          NETCDF =1(on; default)`
 
-Example: `DEBUG=1 NETCDF=1 make`
+Example: `DEBUG=1 NETCDF=1 make -C src`
 
 Compile, edit namelist, and run canopy model:
-- `make`
+- `make -C src`
 - `namelist.canopy`
```
Comment (drnimbusrain) [inline README.md]: Should this be input/namelist.canopy?

**Comment by drnimbusrain** [comment]
@zmoon I also checked off adding "logo in readme" as it seems its done?  Do you plan to do the documentation start and CMake in this PR?

**Comment by drnimbusrain** [comment]
@zmoon Other than that and maybe starting to add some version control to this PR, lets get it merged through develop--> main asap!  Nice work!!

**Comment by zmoon** [comment]
I am going to do CMake config in a separate PR, we can set a version then.

**Review by drnimbusrain — APPROVED** [review]

**Comment by drnimbusrain** [comment]
@zmoon This looks good to go, and will merge.  I think it also should be propagated to the stable branch.

**Pull request merged**